File: my-credentials-plugin/src/main/java/holygradle/credentials/Credentials.java
package holygradle.credentials;

/**
 * Simple holder class for username and password.
 */
public final class Credentials {
    public final String userName;
    public final String password;

    public Credentials(String userName, String password) {
        this.userName = userName;
        this.password = password;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Credentials that = (Credentials) o;

        if (password != null ? !password.equals(that.password) : that.password != null) return false;
        //noinspection RedundantIfStatement
        if (userName != null ? !userName.equals(that.userName) : that.userName != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = userName != null ? userName.hashCode() : 0;
        result = 31 * result + (password != null ? password.hashCode() : 0);
        return result;
    }
}

File: custom-gradle-core-plugin/src/main/kotlin/holygradle/custom_gradle/plugin_apis/Credentials.kt
package holygradle.custom_gradle.plugin_apis

import org.gradle.api.credentials.PasswordCredentials

/**
 * Simple holder class for username and password.
 * TODO 2019-05-10 HughG: Rename to something like DefaultPasswordCredentials, and maybe make internal.
 */
data class Credentials(
        private var _username: String?,
        private var _password: String?
) : PasswordCredentials {
    override fun getUsername(): String? = _username

    override fun setUsername(userName: String?) {
        _username = userName
    }

    override fun getPassword(): String? = _password

    override fun setPassword(password: String?) {
        _password = password
    }

    override fun toString(): String {
        return "Credentials{" +
                "userName='" + _username + '\''.toString() +
                ", password='********'" +
                '}'.toString()
    }
}
